### PR TITLE
Update filterref by update-device

### PR DESCRIPTION
**Pull request by yalzhang**
Add 2 more scenarios for live update filterref except for add filterref:
update filterref, del filterref.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
depend on https://github.com/avocado-framework/avocado-vt/pull/2021
run the additional cases locally and pass, and no impact for the original cases.
new add cases: 
 (06/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.add_filter: PASS (17.27 s)
 (07/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.update_filter: PASS (19.41 s)
 (08/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.del_filter: PASS (18.11 s)
$ run iface_update
JOB ID     : ac2148b6a996b691e56921eabd454f6297a82330
JOB LOG    : /root/avocado/job-results/job-2019-03-28T03.02-ac2148b/job.log
 (01/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.update_bandwidth: PASS (22.01 s)
 (02/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.update_link: PASS (60.00 s)
 (03/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.update_link_with_rom: PASS (67.08 s)
 (04/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.update_link_without_addr: PASS (62.58 s)
 (05/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.update_source: PASS (17.28 s)
 (06/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.add_filter: PASS (17.27 s)
 (07/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.update_filter: PASS (19.41 s)
 (08/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.del_filter: PASS (18.11 s)
 (09/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.positive_test.cold_update_alias: PASS (13.92 s)
 (10/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_alias: PASS (15.32 s)
 (11/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_queues: PASS (16.06 s)
 (12/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_driver: PASS (15.35 s)
 (13/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_model: PASS (15.92 s)
 (14/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_rom: PASS (15.39 s)
 (15/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.add_rom_enable: PASS (15.39 s)
 (16/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_rom_enable: PASS (16.13 s)
 (17/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_rom_and_address: PASS (16.20 s)
 (18/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.del_rom_enable: FAIL: Run '/bin/virsh update-device avocado-vt-vm1 /var/tmp/xml_utils_temp_oas26y_h.xml ' expect fail, but run successfully. (16.11 s)
----> there is a bug on rhel8 currently
 (19/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_bandwidth: PASS (15.77 s)
 (20/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_target: PASS (15.42 s)
 (21/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_address: PASS (15.44 s)
 (22/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_mtu.not_support: PASS (20.78 s)
 (23/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_mtu.invalid_number: PASS (20.65 s)
 (24/24) type_specific.io-github-autotest-libvirt.virtual_network.iface_update.negative_test.update_mtu.invalid_character: PASS (20.82 s)
RESULTS    : PASS 23 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 555.04 s

File: libvirt/tests/src/virtual_network/iface_update.py [hunk 1]
```diff
@@ -174,6 +176,11 @@ def check_mtu():
                 else:
                     test.fail("Get filter %s is not equal to set %s"
                               % (iface_filter_value, new_iface_filter))
+            if del_filter:
+                iface_filter_value = iface_aft.find('filterref')
+                if iface_filter_value is not None:
```
Comment (chloerh) [inline libvirt/tests/src/virtual_network/iface_update.py]: if iface_filter_value:

File: libvirt/tests/src/virtual_network/iface_update.py [hunk 2]
```diff
@@ -174,6 +176,11 @@ def check_mtu():
...
+                    logging.debug("After delete, the filter still exists: %s"
+                                  % iface_filter_value)
```
Comment (chloerh) [inline libvirt/tests/src/virtual_network/iface_update.py]: , instead of %

File: libvirt/tests/src/virtual_network/iface_update.py (hunk 1 above)
Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_update.py]: updated, Thank you!

File: libvirt/tests/src/virtual_network/iface_update.py (hunk 2 above)
Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_update.py]: updated, Thank you!

**Comment by chloerh** [comment]
LGTM

**Comment by chloerh** [comment]
LGTM

**Pull request merged**